test/python/quantum_info/operators/test_measures.py

```diff
@@ -184,6 +187,37 @@ def test_diamond_norm(self, num_qubits):
         except cvxpy.SolverError:
             self.skipTest("CVXPY solver failed.")
 
+    def test_unitary_diamond_distance(self):
+        """Test the unitary_diamond_distance function for RZGates
+        with a specific set of angles."""
+        angles = np.linspace(0, 2 * np.pi, 10, endpoint=False)
+        for angle in angles:
+            op1 = Operator(RZGate(angle))
+            op2 = Operator.from_label("I")
+            d2 = np.cos(angle / 2) ** 2  # analytical formula for hull distance
+            target = np.sqrt(1 - d2) * 2
+            self.assertAlmostEqual(unitary_diamond_distance(op1, op2), target, places=7)
+
+    @combine(num_qubits=[1, 2, 3])
+    def test_unitary_diamond_distance_random(self, num_qubits):
+        """Tests the unitary_diamond_distance for random unitaries.
+        Compares results with semi-definite program."""
+        try:
+            import cvxpy
+        except ImportError:
+            # Skip test if CVXPY not installed
+            self.skipTest("CVXPY not installed.")
+        op1 = random_unitary(2**num_qubits)
+        op2 = random_unitary(2**num_qubits)
+        choi1 = Choi(op1)
+        choi2 = Choi(op2)
+        delta_choi = choi1 - choi2
+        try:
+            target = diamond_norm(delta_choi)
+            self.assertAlmostEqual(unitary_diamond_distance(op1, op2), target, places=4)
+        except cvxpy.SolverError:
+            self.skipTest("CVXPY solver failed.")
```

Comment:
Remove the `try ... except`. Instead, use a seeded random number generator (e.g. `np.random.default_rng(1234)`) and choose the seed so that cvxpy succeeds every time.

Reply:
It may be worth leaving the try catch statement. Looking at how QuTiP implements their `d_norm` tests, we can see the following.

First they comment:
```
# dnorm tests have always been slightly flaky; in some cases, cvxpy will fail
# to solve the problem, and this can cause an entire test-suite failure.  As
# long as we are using random tests (perhaps not ideal), this will happen
# occasionally.  This isn't entirely a bug, it's just a reality of using a
# one-size-fits-all solver; we've historically assumed users who come up
# against this sort of thing will be accepting of the fact that dnorm
# calculation is nontrivial, and isn't always entirely feasible.
#
# To deal with it, we allow each test to be rerun twice, using
# pytest-rerunfailures.  This should forbid pathological cases where the test
# is failing every time, but not penalise one-off failures.  As far as we know,
# the failing tests always involve a random step, so triggering a re-run will
# have them choose new variables as well.
```

Also they skip tests if the platform used is windows:
```
@pytest.mark.skipif("Windows" in platform.system(),
                    reason="Skipping dnorm tests -- cvxpy and cvxopt can be"
                    " installed on Windows but fail to correctly solve"
                    " the given optimization problems.")
```

Finally, note that in the existing implementation of `test_diamond_norm`, there is a similar try catch statement for `cvxpy.SolverError` (see line 167). Maybe this is in place for the reasons discussed above. Either way, we should make sure both tests have consistent implementations.

Reply:
Fwiw I wrote that comment 3 years ago before I started at IBM, and I didn't really know very much about testing strategy back then. Assuming (as my comment implies) `cvxpy` has some uncontrollable internal randomness in its algorithm, it's perhaps worth leaving this skip, but if we can make the algorithm fully deterministic across Python runs by seeding everything, it'll be safer to do that.

Reply:
Also about Windows: cvxpy updated its solvers _a lot_ since that test skip in QuTiP was written - it may not be accurate anymore.

Reply:
Wow! What a coincidence. In that case, I'll just seed everything and remove the try catch. It looks like the CI pipeline tests in windows anyways. 

What's the best approach here? 
- Merge the PR *without* the try catch and patch it if we get unexpected test failures 
- Or keep the try catch in

Sadly, I can't seem to install cvxpy on my machine (unless I'm using anaconda) so not sure how much experimentations I can do on my end.

Reply:
On an unrelated note, do you know how I can join the qiskit slack? I think IBM must have updated something because I swear I used to be part of the workspace

Reply:
>Merge the PR without the try catch and patch it if we get unexpected test failures

Let's go with this.

Re: Slack. Can you try logging in at http://qiskit.slack.com/?

Reply:
Sounds good, have made the change. 

Note: as we're no longer using the cvxpy import (for the cvxpy solver error), I has to add a `# pylint: disable=import-error` comment on those lines. Is this good practice?